crates/http-api-bindings/src/embedding/rate_limit.rs

```diff
@@ -0,0 +1,63 @@
+use std::{
+    sync::Arc,
+    task::{Context, Poll},
+    time,
+};
+
+use async_trait::async_trait;
+use futures::future::BoxFuture;
+use tabby_inference::Embedding;
+use tokio::sync::Mutex;
+use tower::{Service, ServiceBuilder, ServiceExt};
+
+struct EmbeddingService {
+    embedding: Arc<dyn Embedding>,
```

Comment:
can this being `embedding: T` / `Box<dyn Embedding>`

Reply:
will follow up in next release